ibtk/src/math/fortran/vclaplace2d.f.m4

```diff
@@ -18,31 +18,100 @@ include(SAMRAI_FORTDIR/pdat_m4arrdim2d.i)dnl
 c
 ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 c
-c     Compute the arithmetic average of four inputs
+c     Compute the arithmetic and harmonic average based on the number of inputs
 c
 ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-c
-      REAL function a_avg(a0,a1,a2,a3)
-      implicit none
-      REAL a0,a1,a2,a3
-      a_avg = 0.25d0*(a0+a1+a2+a3)
-      return
-      end
-c
-ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-c
-c     Compute the harmonic average of four inputs
-c
-ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-c
-      REAL function h_avg(a0,a1,a2,a3)
-      implicit none
-      REAL a0,a1,a2,a3
-      REAL dmr
-      dmr = 1.d0/a0+1.d0/a1+1.d0/a2+1.d0/a3
-      h_avg = 4.d0/dmr
-      return
-      end
+      module average
```

Comment:
Since we use de-facto fortran 77 everywhere else I would prefer to use the old-fashioned version of function declarations for style uniformity.

In general, modules are much better, but unless we want to convert everything I don't think it makes sense to mix and match right now.

Reply:
There is no other way to do function overloading in Fortran. We searched and found only this. We would like to keep this.

Reply:
This is true - one workaround would be to not do function overloading and instead either define several independent functions or one function which takes an array and array length as argument.

Independently of the style considerations, CMake requires that Fortran module dependencies be handled in a somewhat explicit way. Most of the CI configurations cannot find the `.mod` files right now (GCC 12 appears to work by coincidence only). Similarly, we cannot compile this with autotools for similar reasons.

Since we don't really compile Fortran but instead preprocess m4 files into Fortran, specifying these dependencies is somewhat complex and is another reason that we should use the old-fashioned functions declarations. No matter what solution we work with it needs to work with the configurations we test in the CI.

Reply:
I see. Let's then use several independent functions. Modules don't seem to be worth the trouble, especially if their compilation `.mod` is handled differently by different compilers.